expan/core/results.py

```diff
@@ -0,0 +1,117 @@
+    :param ci: a dict where keys are percentiles and values are the corresponding value for the statistic.
+    :type  ci: dict
+    :type stop: bool
+    """
+    def __init__(self, control_statistics, treatment_statistics, delta, ci, p, statistical_power, stop):
+        super(EarlyStoppingTestStatistics, self).__init__(control_statistics, treatment_statistics, delta, ci, p, statistical_power)
+        self.stop = stop
+
+
+class CorrectedTestStatistics(JsonSerializable):
+    """ Holds original and corrected statistics. This class should be used to hold statistics for multiple testing.
+    original_test_statistics and corrected_test_statistics should have the same type.
+    
+    :param original_test_statistics: test result before correction
+    :type  original_test_statistics: SimpleTestStatistics or EarlyStoppingTestStatistics
+    :param corrected_test_statistics: test result after correction
+    :type  corrected_test_statistics: SimpleTestStatistics or EarlyStoppingTestStatistics
+    """
+    def __init__(self, original_test_statistics, corrected_test_statistics):
+        type1 = type(original_test_statistics)
+        type2 = type(corrected_test_statistics)
+        if type1 != type2:
+            raise RuntimeError("Type mismatch for type " + str(type1) + " and " + str(type2))
+        if not isinstance(original_test_statistics, BaseTestStatistics):
+            raise RuntimeError("Input should be instances of BaseTestStatistics or its subclass")
+        self.original_test_statistics  = original_test_statistics
+        self.corrected_test_statistics = corrected_test_statistics
+
+
+# --------- Below are the data structure of test results --------- #
+class StatisticalTestResult(JsonSerializable):
+    """ This class holds the results of a single statistical test.
+    
+    :param test: information about the statistical test
+    :type  test: StatisticalTest
+    :param result: result of this statistical test
+    :type  result: BaseTestStatistics or its subclasses or CorrectedTestStatistics  #TODO: better approach?
+    """
+    def __init__(self, test, result):
+        self.test   = test
+        self.result = result
+
+
+class MultipleTestSuiteResult(JsonSerializable):
+    """ This class holds the results of a MultipleTestSuite.
+    
+    :param statistical_test_results: test results for all statistical testing unit
+    :type  statistical_test_results: list[StatisticalTestResult]
+    :param correction_method: method used for multiple testing correction. Possible values are:
+                              "none": no correction
+                              "bh": benjamini hochberg correction
+                              "bf": bonferroni correction
+    :type  correction_method: str
+    """
+    def __init__(self, statistical_test_results, correction_method="none"):
+        self.statistical_test_results = statistical_test_results
+        if correction_method not in ["none", "bh", "bf"]:
```

Comment:
Why not stick to something like an `Enum`?

Reply:
So, for analysis service it's ok to have parameters for correction as 0, 1 or 2. We wanted to leave at least a short names for correction for somebody who can use expan from outside Zalando. I also don't think we need to have a class here. Or you have some advantages in mind for doing it as Enum?

Reply:
Each time you add a new correction type or change an existing one, you'll have to find all occurences of, say, `'bh'` in the code. A dedicated `Enum`-type class would be the unique source of truth.

Reply:
@gbordyugov yes I agree. One more question: do you also think we need to create a Enum for the delta methods? We currently maintain a lookup table for "group_sequential", "bayes_factor", "fixed_horizon", etc

Reply:
Yes, that would be another application of this strategy, totally agree!